### ISSUE TITLE
Disabling firewalld should not be the goto method, instead open up the necessary ports on firewall

### ISSUE THREAD

**Issue by whowutwut** [IBM/ibm-spectrum-scale-install-infra#12]
The task of disabling the firewall should not be what we do, instead, keep the firewall running if it's there and open up the correct ports required for Scale to function

**Comment by acch** [comment]
Hi @whowutwut! I'm working my way through some of the older issues and found this one...

I'd argue that adjusting the firewall config in every possible deployment scenario is a complex endeavor. I would claim that it's outside of the scope of this project to deal with the exact configuration steps necessary in a given scenario. Some might be relying on `firewall-cmd`, others might deploy old-fashioned `iptables`, just to name two RHEL examples...

Hence, I'd say that we provide a very generic (maybe crude) way to open the required ports, and that is by simply disabling firewalld alltogether. If users need something more fine-grained then they'd need to define their own tasks for doing so. What do you think?

The documentation has the following ([here](https://github.com/acch/ibm-spectrum-scale-install-infra/blob/docs/VARIABLES.md)):

```
- scale_prepare_disable_firewall
  - example: true
  - default: false

  Whether or not to disable Linux firewalld - if you need to keep firewalld active then change this variable to false and apply your custom firewall rules prior to running this role (e.g. as pre_tasks).
```

Do you agree?

**Comment by janfrode** [comment]
My lazy solution has beeen to assume that all members in the cluster is a unit — if you can hack one, you can hack all — then a firewall opening up for ANY from clustermember to clustermember is a good policy.

I typically then install a firewalld policy like this for each host on all hosts:

firewalld-cmd -–zone=public  -–add-rich-rule=‘rule family="ipv4" source address=173.16.94.12/32 accept’

**Comment by janfrode** [comment]
I don't think this should be too difficult to solve for both iptables and firewalld, but firewalld is nicest.. We could define one "zone" for the internal cluster communication:

**Comment by janfrode** [comment]
So here's s first try:

https://github.com/janfrode/ibm-spectrum-scale-install-infra/tree/firewalld

Any comments will be appreciated.

**Comment by janfrode** [comment]
Added automatically adding addresses to ipset in https://github.com/janfrode/ibm-spectrum-scale-install-infra/commit/f25ab2d57e27e054a1de6171e1aa6833911c8186
